src/context/dataContext.js

```diff
@@ -0,0 +1,33 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { dataReducer, initialState } from "../redux/dataReducer";
+import axios from "axios";
+
+const DataContext = createContext({
+  dataState: {},
+  dataDispatch: () => {},
+});
+
+const DataProvider = ({ children }) => {
+  const [dataState, dataDispatch] = useReducer(dataReducer, initialState);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/products");
```

Comment:
Since you are fetching data inside your context, your products are getting loaded while the user is in the landing page itself which I don't think is a nice idea, products should be fetched only when the user is in the landing page. You can use useEffect combined with useState in your productListing page and then also you can get the data.